File: reacjilator-java-jslack/src/main/java/slackapp_backend/service/SlackWebApi.java
package slackapp_backend.service;

import com.github.seratch.jslack.Slack;
import com.github.seratch.jslack.api.methods.SlackApiException;
import com.github.seratch.jslack.api.methods.request.chat.ChatPostMessageRequest;
import com.github.seratch.jslack.api.methods.request.conversations.ConversationsRepliesRequest;
import com.github.seratch.jslack.api.methods.response.chat.ChatPostMessageResponse;
import com.github.seratch.jslack.api.methods.response.conversations.ConversationsRepliesResponse;

import java.io.IOException;

public class SlackWebApi {

    private final Slack slack = Slack.getInstance();
    private final String token = System.getenv("SLACK_API_TOKEN");

    public ConversationsRepliesResponse fetchConversationsReplies(String channelId, String threadTs) throws IOException, SlackApiException {
        return slack.methods().conversationsReplies(ConversationsRepliesRequest.builder()
                .token(token)
                .channel(channelId)
                .ts(threadTs)
                .build());
    }

    public ChatPostMessageResponse postMessage(String channelId, String threadTs, String text) throws IOException, SlackApiException {
        return slack.methods().chatPostMessage(ChatPostMessageRequest.builder()
                .token(token)
                .channel(channelId)
                .threadTs(threadTs)
                .text(text)
                .username("Reacjilator Bot")
                .build());
    }

}

File: reacjilator-scala-jslack/src/main/scala/slackapp/service/SlackWebApi.scala
package slackapp.service

import com.github.seratch.jslack.Slack
import com.github.seratch.jslack.api.methods.request.chat.ChatPostMessageRequest
import com.github.seratch.jslack.api.methods.request.conversations.ConversationsRepliesRequest
import com.github.seratch.jslack.api.methods.response.chat.ChatPostMessageResponse
import com.github.seratch.jslack.api.methods.response.conversations.ConversationsRepliesResponse

class SlackWebApi {
  import SlackWebApi._

  def fetchConversationsReplies(channelId: String, threadTs: String): ConversationsRepliesResponse = {
    slack.methods.conversationsReplies(
      ConversationsRepliesRequest.builder
        .token(token)
        .channel(channelId)
        .ts(threadTs)
        .build
    )
  }

  def postMessage(channelId: String, threadTs: String, text: String): ChatPostMessageResponse = {
    slack.methods.chatPostMessage(
      ChatPostMessageRequest.builder
        .token(token)
        .channel(channelId)
        .threadTs(threadTs)
        .text(text)
        .username("Reacjilator Bot")
        .build
    )
  }

}

object SlackWebApi {

  private val slack = Slack.getInstance
  private val token = sys.env("SLACK_API_TOKEN")

}
